### ISSUE TITLE
FAILED !trtexec --onnx=converted.onnx \--saveEngine=engine.trt \--useCudaGraph

### ISSUE THREAD

**Issue by cory619-seungwon** [NVIDIA/TensorRT#3649]
## Description

 I tried to run Mask-RCNN on GPU, but, it fails to build engine.trt with error below... When running without tensorRT, my model(Mask_RCNN) works properly...

**The code is as below.**
(build TensorRT engine [engine.trt] - Float32)
!trtexec --onnx=converted.onnx \
    --saveEngine=engine.trt \
    --useCudaGraph 

**The error is as below.**
&&&& RUNNING TensorRT.trtexec [TensorRT v8601] # trtexec --onnx=converted.onnx \--saveEngine=engine.trt -- \ useCudaGraph
[02/01/2024-18:00:13] [I] === Model Options ===
[02/01/2024-18:00:13] [I] Format: ONNX
[02/01/2024-18:00:13] [I] Model: converted.onnx
[02/01/2024-18:00:13] [I] Output:
[02/01/2024-18:00:13] [I] === Build Options ===
[02/01/2024-18:00:13] [I] Max batch: explicit batch
[02/01/2024-18:00:13] [I] Memory Pools: workspace: 2000 MiB, dlaSRAM: default, dlaLocalDRAM: default, dlaGlobalDRAM: default
[02/01/2024-18:00:13] [I] minTiming: 1
[02/01/2024-18:00:13] [I] avgTiming: 8
[02/01/2024-18:00:13] [I] Precision: FP32
[02/01/2024-18:00:13] [I] LayerPrecisions: 
[02/01/2024-18:00:13] [I] Layer Device Types: 
[02/01/2024-18:00:13] [I] Calibration: 
[02/01/2024-18:00:13] [I] Refit: Disabled
[02/01/2024-18:00:13] [I] Version Compatible: Disabled
[02/01/2024-18:00:13] [I] TensorRT runtime: full
[02/01/2024-18:00:13] [I] Lean DLL Path: 
[02/01/2024-18:00:13] [I] Tempfile Controls: { in_memory: allow, temporary: allow }
[02/01/2024-18:00:13] [I] Exclude Lean Runtime: Disabled
[02/01/2024-18:00:13] [I] Sparsity: Disabled
[02/01/2024-18:00:13] [I] Safe mode: Disabled
[02/01/2024-18:00:13] [I] Build DLA standalone loadable: Disabled
[02/01/2024-18:00:13] [I] Allow GPU fallback for DLA: Disabled
[02/01/2024-18:00:13] [I] DirectIO mode: Disabled
[02/01/2024-18:00:13] [I] Restricted mode: Disabled
[02/01/2024-18:00:13] [I] Skip inference: Disabled
[02/01/2024-18:00:13] [I] Save engine: engine.trt
[02/01/2024-18:00:13] [I] Load engine: 
[02/01/2024-18:00:13] [I] Profiling verbosity: 0
[02/01/2024-18:00:13] [I] Tactic sources: Using default tactic sources
[02/01/2024-18:00:13] [I] timingCacheMode: local
[02/01/2024-18:00:13] [I] timingCacheFile: 
[02/01/2024-18:00:13] [I] Heuristic: Disabled
[02/01/2024-18:00:13] [I] Preview Features: Use default preview flags.
[02/01/2024-18:00:13] [I] MaxAuxStreams: -1
[02/01/2024-18:00:13] [I] BuilderOptimizationLevel: -1
[02/01/2024-18:00:13] [I] Input(s)s format: fp32:CHW
[02/01/2024-18:00:13] [I] Output(s)s format: fp32:CHW
[02/01/2024-18:00:13] [I] Input build shapes: model
[02/01/2024-18:00:13] [I] Input calibration shapes: model
[02/01/2024-18:00:13] [I] === System Options ===
[02/01/2024-18:00:13] [I] Device: 0
[02/01/2024-18:00:13] [I] DLACore: 
[02/01/2024-18:00:13] [I] Plugins:
[02/01/2024-18:00:13] [I] setPluginsToSerialize:
[02/01/2024-18:00:13] [I] dynamicPlugins:
[02/01/2024-18:00:13] [I] ignoreParsedPluginLibs: 0
[02/01/2024-18:00:13] [I] 
[02/01/2024-18:00:13] [I] === Inference Options ===
[02/01/2024-18:00:13] [I] Batch: Explicit
[02/01/2024-18:00:13] [I] Input inference shapes: model
[02/01/2024-18:00:13] [I] Iterations: 10
[02/01/2024-18:00:13] [I] Duration: 3s (+ 200ms warm up)
[02/01/2024-18:00:13] [I] Sleep time: 0ms
[02/01/2024-18:00:13] [I] Idle time: 0ms
[02/01/2024-18:00:13] [I] Inference Streams: 1
[02/01/2024-18:00:13] [I] ExposeDMA: Disabled
[02/01/2024-18:00:13] [I] Data transfers: Enabled
[02/01/2024-18:00:13] [I] Spin-wait: Disabled
[02/01/2024-18:00:13] [I] Multithreading: Disabled
[02/01/2024-18:00:13] [I] CUDA Graph: Enabled
[02/01/2024-18:00:13] [I] Separate profiling: Disabled
[02/01/2024-18:00:13] [I] Time Deserialize: Disabled
[02/01/2024-18:00:13] [I] Time Refit: Disabled
[02/01/2024-18:00:13] [I] NVTX verbosity: 0
[02/01/2024-18:00:13] [I] Persistent Cache Ratio: 0
[02/01/2024-18:00:13] [I] Inputs:
[02/01/2024-18:00:13] [I] === Reporting Options ===
[02/01/2024-18:00:13] [I] Verbose: Disabled
[02/01/2024-18:00:13] [I] Averages: 10 inferences
[02/01/2024-18:00:13] [I] Percentiles: 90,95,99
[02/01/2024-18:00:13] [I] Dump refittable layers:Disabled
[02/01/2024-18:00:13] [I] Dump output: Disabled
[02/01/2024-18:00:13] [I] Profile: Disabled
[02/01/2024-18:00:13] [I] Export timing to JSON file: 
[02/01/2024-18:00:13] [I] Export output to JSON file: 
[02/01/2024-18:00:13] [I] Export profile to JSON file: 
[02/01/2024-18:00:13] [I] 
[02/01/2024-18:00:13] [I] === Device Information ===
[02/01/2024-18:00:13] [I] Selected Device: NVIDIA RTX A5000
[02/01/2024-18:00:13] [I] Compute Capability: 8.6
[02/01/2024-18:00:13] [I] SMs: 64
[02/01/2024-18:00:13] [I] Device Global Memory: 24563 MiB
[02/01/2024-18:00:13] [I] Shared Memory per SM: 100 KiB
[02/01/2024-18:00:13] [I] Memory Bus Width: 384 bits (ECC disabled)
[02/01/2024-18:00:13] [I] Application Compute Clock Rate: 1.695 GHz
[02/01/2024-18:00:13] [I] Application Memory Clock Rate: 8.001 GHz
[02/01/2024-18:00:13] [I] 
[02/01/2024-18:00:13] [I] Note: The application clock rates do not reflect the actual clock rates that the GPU is currently running at.
[02/01/2024-18:00:13] [I] 
[02/01/2024-18:00:13] [I] TensorRT version: 8.6.1
[02/01/2024-18:00:13] [I] Loading standard plugins
[02/01/2024-18:00:13] [I] [TRT] [MemUsageChange] Init CUDA: CPU +82, GPU +0, now: CPU 11008, GPU 1281 (MiB)
[02/01/2024-18:00:23] [I] [TRT] [MemUsageChange] Init builder kernel library: CPU +1448, GPU +266, now: CPU 13591, GPU 1547 (MiB)
[02/01/2024-18:00:23] [I] Start parsing network model.
[02/01/2024-18:00:24] [I] [TRT] ----------------------------------------------------------------
[02/01/2024-18:00:24] [I] [TRT] Input filename:   converted.onnx
[02/01/2024-18:00:24] [I] [TRT] ONNX IR version:  0.0.6
[02/01/2024-18:00:24] [I] [TRT] Opset version:    11
[02/01/2024-18:00:24] [I] [TRT] Producer name:    
[02/01/2024-18:00:24] [I] [TRT] Producer version: 
[02/01/2024-18:00:24] [I] [TRT] Domain:           
[02/01/2024-18:00:24] [I] [TRT] Model version:    0
[02/01/2024-18:00:24] [I] [TRT] Doc string:       
[02/01/2024-18:00:24] [I] [TRT] ----------------------------------------------------------------
[02/01/2024-18:00:24] [I] Finished parsing network model. Parse time: 0.470887
&&&& FAILED TensorRT.trtexec [TensorRT v8601] # trtexec --onnx=converted.onnx --saveEngine=engine.trt --useCudaGraph --workspace=2000
[02/01/2024-18:00:13] [W] --workspace flag has been deprecated by --memPoolSize flag.
[02/01/2024-18:00:24] [W] [TRT] onnx2trt_utils.cpp:374: Your ONNX model has been generated with INT64 weights, while TensorRT does not natively support INT64. Attempting to cast down to INT32.
[02/01/2024-18:00:24] [E] Error[4]: [shapeContext.cpp::nvinfer1::builder::ShapeContext::op::1942] Error Code 4: Shape Error (broadcast dimensions must be conformable)
[02/01/2024-18:00:24] [E] [TRT] ModelImporter.cpp:771: While parsing node number 134 [Add -> "/backbone/Add_1_output_0"]:
[02/01/2024-18:00:24] [E] [TRT] ModelImporter.cpp:772: --- Begin node ---
[02/01/2024-18:00:24] [E] [TRT] ModelImporter.cpp:773: input: "/backbone/fpn_lateral3/Conv_output_0"
input: "/backbone/Resize_1_output_0"
output: "/backbone/Add_1_output_0"
name: "/backbone/Add_1"
op_type: "Add"

[02/01/2024-18:00:24] [E] [TRT] ModelImporter.cpp:774: --- End node ---
[02/01/2024-18:00:24] [E] [TRT] ModelImporter.cpp:777: ERROR: ModelImporter.cpp:195 In function parseGraph:
[6] Invalid Node - /backbone/Add_1
[shapeContext.cpp::nvinfer1::builder::ShapeContext::op::1942] Error Code 4: Shape Error (broadcast dimensions must be conformable)
[02/01/2024-18:00:24] [E] Failed to parse onnx file
[02/01/2024-18:00:24] [E] Parsing model failed
[02/01/2024-18:00:24] [E] Failed to create engine from model or file.
[02/01/2024-18:00:24] [E] Engine set up failed

## Environment

**TensorRT Version**: 8.6

**NVIDIA GPU**: NVIDIA GTX A5000

**NVIDIA Driver Version**: 531.14

**CUDA Version**: v11.6

**CUDNN Version**: v8.6

Operating System: Window10

Python Version (if applicable): Python 3.8

Tensorflow Version (if applicable):

PyTorch Version (if applicable):

Baremetal or Container (if so, version):

## Relevant Files

**Model link**:

## Steps To Reproduce

**Commands or scripts**:

**Have you tried [the latest release](https://developer.nvidia.com/tensorrt)?**:

**Can this model run on other frameworks?** For example run ONNX model with ONNXRuntime (`polygraphy run <model.onnx> --onnxrt`):

**Comment by RajUpadhyay** [comment]
It is showing shape error. Did you try to check your onnx with netron?
Did you set the MIN_SIZE_TEST and MAX_SIZE_TEST to 1344 when you first created onnx from detectron2?

Since detectron2 has a dynamic shape, you need to set it to 1344 and then use create_onnx.py to get your converted.onnx.

**Comment by cory619-seungwon** [comment]
@RajUpadhyay Thanks for the reply. I sovled it to modify sample image size as 1344 x 1344. Previous sample image size was 1515 x 1217. The reason why setting image size as 1344 is to follow TensorRT guide line on "https://github.com/NVIDIA/TensorRT/tree/release/8.6/samples/python/detectron2". In that page, there is an explanation: "TensorRT plug-ins used for conversion of this model do not support dynamic shapes, as a result we have to set both height and width of the input tensor to 1344.". 

I think TensorRT for detectron2 dose not support dynamic shape.

Thanks.

**Comment by RajUpadhyay** [comment]
Yes that is correct, if you want a more in depth reason for 1344, follow [this](https://github.com/NVIDIA/TensorRT/issues/2142#:~:text=I%20can%20tell%20you%20the%20answer%20but%20it%20will%20be%20a%20long%20one.) link.

**Comment by zerollzeng** [comment]
close this issue.
